packages/flutter/lib/src/widgets/widget_inspector.dart

```diff
@@ -2885,7 +2884,7 @@ class _WidgetInspectorState extends State<WidgetInspector>
 }
 
 /// Mutable selection state of the inspector.
-class InspectorSelection {
+class InspectorSelection with ChangeNotifier {
```

Comment:
clever. this is a simpler change than making InspectorSelection immutable.

Reply:
it is probably worth adding a test to verify that you are notifying of changes for all the methods that should.